store/root.js

```diff
@@ -73,8 +89,8 @@ class Root {
     this.retrievePluginConfig()
 
     const url = `/data-providers/${this.dataProvider.id}`
-    this.works = new Works(url)
-    this.depositDates = new DepositDates(url)
+    this.works = new Works(url, this.apiRequest)
```

Comment:
It's an options parameter, not a second initialisation parameter.

```suggestion
    this.works = new Works(url, this.options)
```

Despite of that, I propose to simply call it `request`

Reply:
Maybe [passing `this`](https://mobx.js.org/best/store.html#combining-multiple-stores) would be even better. Obviously the serialization would be harder but we don't need it now anyway.